packages/twenty-server/src/modules/workflow/workflow-executor/utils/variable-resolver.util.ts

```diff
@@ -0,0 +1,76 @@
+import { isNil, isString } from '@nestjs/common/utils/shared.utils';
+
+import Handlebars from 'handlebars';
+
+import {
+  WorkflowExecutorException,
+  WorkflowExecutorExceptionCode,
+} from 'src/modules/workflow/workflow-executor/exceptions/workflow-executor.exception';
+
+const VARIABLE_PATTERN = RegExp('\\{\\{(.*?)\\}\\}', 'g');
+
+export const resolve = (
+  unresolvedInput: any,
+  context: Record<string, object>,
+) => {
+  if (isNil(unresolvedInput)) {
+    return unresolvedInput;
+  }
+
+  if (isString(unresolvedInput)) {
+    return resolveString(unresolvedInput, context);
+  }
+
+  const resolvedInput = unresolvedInput;
+
+  if (Array.isArray(unresolvedInput)) {
+    for (let i = 0; i < unresolvedInput.length; ++i) {
+      resolvedInput[i] = resolve(unresolvedInput[i], context);
+    }
+  } else if (typeof unresolvedInput === 'object') {
+    const entries = Object.entries(unresolvedInput);
+
+    for (const [key, value] of entries) {
+      resolvedInput[key] = resolve(value, context);
+    }
+  }
+
+  return resolvedInput;
+};
+
+const resolveString = (
+  input: string,
+  context: Record<string, object>,
+): string => {
+  const matchedTokens = input.match(VARIABLE_PATTERN);
+
+  if (!matchedTokens || matchedTokens.length === 0) {
+    return input;
+  }
+
+  if (matchedTokens.length === 1 && matchedTokens[0] === input) {
+    return evalFromContext(input, context);
+  }
+
+  return input.replace(VARIABLE_PATTERN, (matchedToken, _) => {
+    const processedToken = evalFromContext(matchedToken, context);
+
+    return processedToken;
+  });
```

Comment:
Aren't we redoing the job of Handlebars here? Couldn't we call `evalFromContext` on the whole `input` string and let Handlebars replace the variables? i might misunderstand something.

Reply:
I want to focus on variables only there. Giving the whole string to handlebars can make it translate pretty much anything. See different templates here https://handlebarsjs.com/guide/#nested-input-objects